src/NServiceBus.Testing.Tests/Sagas/ExternalHandlerSimulator.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace NServiceBus.Testing.Tests.Sagas
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class ExternalHandlerSimulator
+    {
+        [Test]
+        public async Task ExternalHandlerShouldSupplyReplyMessage()
+        {
+            var testableSaga = new TestableSaga<MySaga, MyData>();
+
+            testableSaga.SimulateHandlerResponse<RunStep1Command, Step1Response>(cmd => new Step1Response());
+
+            var startResult = await testableSaga.Process(new StartMsg { CorrId = "abc" });
+
+            Assert.That(startResult.Context.SentMessages.Count, Is.EqualTo(1));
+            Assert.That(testableSaga.QueueLength, Is.EqualTo(1));
+            Assert.That(testableSaga.QueuePeek().Type, Is.EqualTo(typeof(Step1Response)));
+
+            var continueResult = await testableSaga.ProcessQueuedMessage();
```

Comment:
What is the benefit of this compared to just call Process with the response message?

Reply:
There is a method like that, but feels horribly inelegant given you have to also provide the saga id for the auto correlation, something I’d prefer to be inside the black box.

Reply:
Got it, that is clearly a benefit. It made me realize that this style of testing also tests mappings and consecutive message exchanges compared to pure "unit test one handler at a time". I like that you have the entire flow of message for a "scenario" right in front of you in the same test method.

👍

Reply:
I like that word `Scenario`, I had been using the word conversation but that didn't feel right. Would something with Scenario in it be a better name for this than `TestableSaga`? Though `TestableX` is kind of the pattern of everything else in here like `TestableMessageHandlerContext` and the testable saga isn't really a scenario either - your test can test a scenario, but the thing itself is analogous to a deployed saga, with its own storage and able to differentiate between messages from different conversations.

All this to say I hate naming things.